src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -282,8 +342,13 @@ K2PgStatus PgGate_ExecCreateTable(const char *database_name,
                               bool if_not_exist,
                               bool add_primary_key,
                               const std::vector<K2PGColumnDef>& columns) {
-  elog(DEBUG5, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
-  return K2PgStatus::NotSupported;
+    elog(DEBUG5, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
+    auto [status, is_pg_catalog_table, schema] = makeSChema(schema_name, columns);
+    if (!status.ok()) {
+        return status;
+    }
+    const k2pg::PgObjectId table_object_id(database_oid, table_oid);
+    return pg_gate->GetCatalogClient()->CreateTable(database_name, table_name, table_object_id, schema, is_pg_catalog_table, false /* is_shared_table */, if_not_exist);
```

Comment:
is_shared_table needs to be passed from opengauss here and all the way to catalog manager. For a shared table, catalog manager might store it in a different k2 collection from the current k2 database collection. We should not skip it here, or always set it to false.

Reply:
In that case pg gate api signature should be modified to take additional parameter is_shared_table similar to chogori sql https://github.com/futurewei-cloud/chogori-sql/blob/master/src/k2/connector/pggate/pg_gate_api.h#L155. And caller needs to pass this parameter. Let me if I should do that or you plan to do.